src/nvim/option.c

```diff
@@ -3485,36 +3528,254 @@ vimoption_T *get_option(int opt_idx)
+  const char *errmsg = NULL;
+  bool restore_chartab = false;
+  bool free_oldval = (opt->flags & P_ALLOCED);
+  bool value_changed = false;
+
+  opt_did_set_cb_T did_set_cb;
+  optset_T did_set_cb_args = {
+    .os_varp = varp,
+    .os_idx = opt_idx,
+    .os_flags = opt_flags,
+    .os_oldval = old_value.data,
+    .os_newval = new_value.data,
+    .os_value_checked = false,
+    .os_value_changed = false,
+    .os_restore_chartab = false,
+    .os_doskip = false,
+    .os_errbuf = errbuf,
+    .os_errbuflen = errbuflen,
+    .os_buf = curbuf,
+    .os_win = curwin
+  };
+
+  if ((int *)varp == &p_force_on) {
+    did_set_cb = did_set_force_on;
+  } else if ((int *)varp == &p_force_off) {
+    did_set_cb = did_set_force_off;
+  } else {
+    did_set_cb = opt->opt_did_set_cb;
+  }
+
+  // Disallow changing some options from secure mode
+  if ((secure || sandbox != 0) && (opt->flags & P_SECURE)) {
+    errmsg = e_secure;
+    // Check for a "normal" directory or file name in some string options.
+  } else if (new_value.type == kOptValTypeString
+             && check_illegal_path_names(*(char **)varp, opt->flags)) {
+    errmsg = e_invarg;
+  } else if (did_set_cb != NULL) {
+    // Invoke the option specific callback function to validate and apply the new value.
+    errmsg = did_set_cb(&did_set_cb_args);
+    // Whether option should be processed further or skipped.
+    *doskip = did_set_cb_args.os_doskip;
+    // The 'filetype' and 'syntax' option callback functions may change the os_value_changed field.
+    value_changed = did_set_cb_args.os_value_changed;
+    // The 'keymap', 'filetype' and 'syntax' option callback functions may change the
+    // os_value_checked field.
+    *value_checked = did_set_cb_args.os_value_checked;
+    // The 'isident', 'iskeyword', 'isprint' and 'isfname' options may change the character table.
+    // On failure, this needs to be restored.
+    restore_chartab = did_set_cb_args.os_restore_chartab;
+  }
+
+  // If an error is detected, restore the previous value.
+  if (errmsg != NULL) {
+    set_option_varp(opt_idx, varp, old_value, true);
+    // When resetting some values, need to act on it.
+    if (restore_chartab) {
+      (void)buf_init_chartab(curbuf, true);
+    }
+
+    // Unset new_value as it is no longer valid.
+    new_value = NIL_OPTVAL;  // NOLINT(clang-analyzer-deadcode.DeadStores)
+  } else {
+    // Re-assign the new value as its value may get freed or modified by the option callback.
+    new_value = optval_from_varp(opt_idx, varp);
+
+    // Remember where the option was set.
+    set_option_sctx_idx(opt_idx, opt_flags, current_sctx);
+    // Free options that are in allocated memory.
+    // Use "free_oldval", because recursiveness may change the flags (esp. init_highlight()).
+    if (free_oldval) {
+      optval_free(old_value);
+    }
+    opt->flags |= P_ALLOCED;
+
+    // Check the bound for num options.
+    if (new_value.type == kOptValTypeNumber) {
+      errmsg
```

Comment:
strange uncrustify bug, would rather add the newline before `errbuf, errbuflen, errmsg)`

Reply:
fixed